File: app/src/main/java/com/practice/forecast/ui/splash/SplashContract.java
package com.practice.forecast.ui.splash;

import com.practice.forecast.ui.arch.Contract;

import androidx.lifecycle.LiveData;

public interface SplashContract {
    interface BaseSplashViewModel{
        void startTimer();
        LiveData<Boolean> getIsReady();
    }

    interface Host extends Contract.Host {
        void openMapFragment();
    }
}

File: app/src/main/java/com/practice/forecast/ui/splash/SplashContract.kt
package com.practice.forecast.ui.splash

import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.LiveData
import com.practice.forecast.ui.arch.Contract

interface SplashContract {
    interface BaseSplashViewModel: LifecycleObserver {
        fun startTimer()
        fun getIsReady(): LiveData<Boolean>
    }

    interface Host : Contract.Host {
        fun openMapFragment()
    }
}